src/masternodes/rpc_proposals.cpp

```diff
@@ -474,9 +475,26 @@ UniValue votegov(const JSONRPCRequest &request) {
             throw JSONRPCError(RPC_INVALID_PARAMETER,
                                strprintf("Proposal <%s> is not in voting period", propId.GetHex()));
         }
+        if (id.length() == 64) {
+            mnId = ParseHashV(id, "masternodeId");
+        } else {
+            CTxDestination dest = DecodeDestination(id);
+            if (!IsValidDestination(dest)) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER,
+                                    strprintf("The masternode id or address is not valid: %s", id));
+            }
+            const CKeyID ckeyId = dest.index() == PKHashType ? CKeyID(std::get<PKHash>(dest)) : CKeyID(std::get<WitnessV0KeyHash>(dest));
```

Comment:
We need to make sure that the provided address is either P2PKH or a P2WPKH, otherwise if someone passes a different variant then `std::get` will throw a `std::bad_variant_access`.

```
            const CKeyID ckeyId  = dest.index() == PKHashType ? CKeyID(std::get<PKHash>(dest))
                                 : dest.index() == WitV0KeyHashType ? CKeyID(std::get<WitnessV0KeyHash>(dest))
                                 : CKeyID();
```